soh/src/code/z_parameter.c

```diff
@@ -2446,8 +2446,17 @@ u8 Item_Give(PlayState* play, u8 item) {
             Flags_SetItemGetInf(ITEMGETINF_OBTAINED_NUT_UPGRADE_FROM_STAGE);
         }
 
-        if (item >= ITEM_POCKET_EGG) {
-            gSaveContext.ship.quest.data.randomizer.adultTradeItems |= ADULT_TRADE_FLAG(item);
+        if (IS_RANDO) {
+            if (item >= ITEM_POCKET_EGG) {
+                Flags_SetRandomizerInf(item - ITEM_POCKET_EGG + RAND_INF_ADULT_TRADES_HAS_POCKET_EGG);
+            } else if (item == ITEM_LETTER_ZELDA) {
+                //don't care about zelda's letter if it's already been shown to the guard
+                if (!Flags_GetInfTable(INFTABLE_SHOWED_ZELDAS_LETTER_TO_GATE_GUARD)) {
+                    Flags_SetRandomizerInf(RAND_INF_CHILD_TRADES_HAS_LETTER_ZELDA);
+                }
+            } else {
+                Flags_SetRandomizerInf(item - ITEM_WEIRD_EGG + RAND_INF_CHILD_TRADES_HAS_WEIRD_EGG);
+            }
```

Comment:
I don't really feel like I can force this on you in this PR because this is touching code that already existed, but if you're open for it, would you mind moving this to a hook as well? If not it'll have to be done with a wider cleanup effort in the future.